### PR TITLE
[PEPC-Boost][13] Add state interference checks for Goerli

### PR DESCRIPTION
**Pull request by bharath-123**
## 📝 Summary

For goerli n/w, we are checking if the ToB tx is a uniswap v3 eth/usdc tx. We add state interference checks to check if the ToB tx is a uniswap v3 eth/usdc tx. In this PR, we add types and methods required to prepare for this check.

It works in the same as mentioned in https://github.com/bharath-123/pepc-boost-relay/pull/15 

For goerli, we check if the tx is a valid uniswap v3 eth/usdc swap. To do this we check if the tx is sent to the swap router with tokenIn and tokenOut as the Eth and Usdc token addresses.

### CODE REVIEW

File: contracts/uniswap-v3-swap-router.go
Comment (michaelneuder) [inline contracts/uniswap-v3-swap-router.go:1]: maybe generated files can be in independent PRs ?

**Review by michaelneuder — APPROVED** [review]
this pr seems to be mostly the generated code, right? seems ok

File: contracts/uniswap-v3-swap-router.go
Comment (bharath-123) [reply contracts/uniswap-v3-swap-router.go:1]: yess!

Comment (bharath-123) [reply contracts/uniswap-v3-swap-router.go:1]: it adds the entire state interference checks! Not just auto generated files. I modularised the entire state interference part! We just to add one go method per n/w for these ToB checks!
